File: apps/s-2-admin/src/api/schedule/Schedule.ts
import { User } from "../user/User";
import { EventType } from "../eventType/EventType";
import { Availability } from "../availability/Availability";

export type Schedule = {
  id: number;
  user?: User;
  name: string;
  timeZone: string | null;
  eventType?: Array<EventType>;
  availability?: Array<Availability>;
};

File: apps/cactus-demo-dotnet/src/Infrastructure/Models/Schedule.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CactusDemoDotnet.Infrastructure.Models;

[Table("Schedules")]
public class ScheduleDbModel
{
    [Key()]
    [Required()]
    public int Id { get; set; }

    public int UserId { get; set; }

    [ForeignKey(nameof(UserId))]
    public UserDbModel User { get; set; } = null;

    [Required()]
    [StringLength(256)]
    public string Name { get; set; }

    [StringLength(256)]
    public string? TimeZone { get; set; }

    public List<EventTypeDbModel>? EventType { get; set; } = new List<EventTypeDbModel>();

    public List<AvailabilityDbModel>? Availability { get; set; } = new List<AvailabilityDbModel>();
}
